### PR TITLE
controller: Change CRUD helper to use httprouter instead of martini

### PR DESCRIPTION
**Pull request by jvatic**
refs #744

Still returns martini handler func for lookup
Moving towards using httphelper pkg for Error and JSON responses

### CODE REVIEW

**Comment by titanous** [comment]
Typo in the commit message `s/marini/martini`.

File: controller/controller.go [hunk 1]
```diff
@@ -112,6 +114,31 @@ func (r *responseHelper) Error(err error) {
...
+		if err == ErrNotFound {
+			rw.WriteHeader(404)
+			return
+		}
+		log.Println(err)
+		respondWithJSON(rw, 500, struct{}{})
+	}
+}
+
+func respondWithJSON(rw http.ResponseWriter, status int, thing interface{}) {
```
Comment (titanous) [inline controller/controller.go]: We have very similar helpers in `pkg/httphelper`, let's standardize on those.

**Comment by jvatic** [comment]
fixed typo

File: controller/controller.go [hunk 2]
```diff
@@ -145,11 +172,17 @@ func appHandler(c handlerConfig) (http.Handler, *martini.Martini) {
...
+	httpRouter := httprouter.New()
+	httpRouter.NotFound = http.HandlerFunc(func(res http.ResponseWriter, req *http.Request) {
+		m.ServeHTTP(res, req)
+	})
+
+	getAppMiddleware := crud("apps", ct.App{}, appRepo, httpRouter)
+	getReleaseMiddleware := crud("releases", ct.Release{}, releaseRepo, httpRouter)
+	getProviderMiddleware := crud("providers", ct.Provider{}, providerRepo, httpRouter)
+	crud("artifacts", ct.Artifact{}, artifactRepo, httpRouter)
+	crud("keys", ct.Key{}, keyRepo, httpRouter)
```
Comment (titanous) [inline controller/controller.go]: It feels like it makes more sense to pass `httpRouter` as the first parameter.

**Comment by jvatic** [comment]
@titanous comments addressed*

*`httphelper.Error` doesn't currently handle `ErrNotFound` or `ct.ValidationError` so we'll need to standardize on something for those.

File: controller/controller.go [hunk 3]
```diff
@@ -112,6 +115,19 @@ func (r *responseHelper) Error(err error) {
 	}
 }
 
+func respondWithError(w http.ResponseWriter, err error) {
```
Comment (titanous) [inline controller/controller.go]: Add a comment noting this is a temporary measure.

**Comment by titanous** [comment]
LGTM pending minor comment

**Pull request merged**